client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java

```diff
@@ -208,4 +210,37 @@ public void existsAliasAsync(GetAliasesRequest getAliasesRequest, ActionListener
         restHighLevelClient.performRequestAsync(getAliasesRequest, Request::existsAlias, RestHighLevelClient::convertExistsResponse,
                 listener, emptySet(), headers);
     }
+
+    /**
+     * Checks if the index (indices) exists or not.
+     * <p>
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-exists.html">
+     * Indices Exists API on elastic.co</a>
+     */
+    public final boolean exists(GetIndexRequest request, Header... headers) throws IOException {
```

Comment:
I made a mistake when I asked you to make these methods final. The class is already final, so now this modifier is redundant which fails the build. Could you run `../../gradlew check` from `client/rest-high-level` to verify that tests run ok?

Reply:
Will do. So far I have only been testing what has been "near" the code which was changed in the PR.